index.js

```diff
@@ -0,0 +1,292 @@
+
+    arrayOfNotes.push (new stickyNote(    
+        note_id,
+        note_name.value,
+        note_text.value,
+        note_dateInput,
+        note_time.value,
+        completed,
+    ));
+
+    saveToLocalStorage("stickyNotes",arrayOfNotes);
+    draw(arrayOfNotes)
+    NOTES_DOM.notes_board.reset();
+
+    }
+
+    // validation functions
+    function DateRegexValidate(date) {
+        let dateRegex = /([12]\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01]))/
+        return dateRegex.test(date)
+    }
+    function timeRegexValidate(time) {
+        let timeRegex = /^(\d{1,2}):(\d{2})(?::(\d{2}))?$/
+        return timeRegex.test(time)
+    }
+    function validateTime(note_time,note_date) {
+        Date.prototype.toDateInputValue = (function() {
+            let local = new Date(this);
+            local.setMinutes(this.getMinutes() - this.getTimezoneOffset());
+            return local.toJSON().slice(11,16);
+        
+        });
+        let timeNow =  new Date().toDateInputValue();
+        timeNow = timeNow.replace(":",'')
+        timeNow = Number(timeNow)
+    
+        let Itime = note_time
+        Itime = Itime.replace(":",'')
+        Itime = Number(Itime)
+
+        Date.prototype.toDateInputValue = (function() {
+            let local = new Date(this);
+            local.setMinutes(this.getMinutes() - this.getTimezoneOffset());
+            return local.toJSON().slice(0,10);
+        
+        
+        });
+
+        let dateNow = new Date().toDateInputValue();
+        dateNow = dateNow.replace("-" , '')
+        dateNow = dateNow.replace("-",'')
+        dateNow = Number(dateNow)
+
+        let Idate = note_date
+        Idate = Idate.replace("-",'')
+        Idate = Idate.replace("-",'')
+        Idate = Number(Idate)
+      
+         if(Idate === dateNow && Itime < timeNow){return false;}
+        
+        
+    }
+    function validateDate(note_date)  {
+        Date.prototype.toDateInputValue = (function() {
+            let local = new Date(this);
+            local.setMinutes(this.getMinutes() - this.getTimezoneOffset());
+            return local.toJSON().slice(0,10);
+        
+        
+        });
+
+        let dateNow = new Date().toDateInputValue();
+        dateNow = dateNow.replace("-" , '')
+        dateNow = dateNow.replace("-",'')
+        dateNow = Number(dateNow)
+
+        let Idate = note_date
+        Idate = Idate.replace("-",'')
+        Idate = Idate.replace("-",'')
+        Idate = Number(Idate)
+      
+         if(Idate < dateNow){ return false;}
+    }
+   // local storage functions 
+
+    function saveToLocalStorage(key, value) {
+        localStorage.setItem(key, JSON.stringify(value));
+      }
+
+      function init() {
+        arrayOfNotes = JSON.parse(localStorage.getItem("stickyNotes")) || [];
+        draw(arrayOfNotes);
+      }
+      init();
+
+function sortBy(value,arr) {
+    switch(value){
+        case ("All"):
```

Comment:
no need for () just case "All":

Reply:
I DONT PUT A FOR () ?? just switch case..